projects/ngx-sub-form/src/lib/one-of.validator.ts

```diff
@@ -0,0 +1,36 @@
+import { AbstractControl } from '@angular/forms';
+import { TypedFormGroup, TypedValidatorFn } from './ngx-sub-form.types';
+import {
+  isNullOrUndefined,
+  OneOfValidatorRequiresMoreThanOneFieldError,
+  OneOfValidatorUnknownFieldError,
+} from './ngx-sub-form-utils';
+
+export namespace NgxSubFormValidators {
+
+  export function oneOf<FormInterface>(keysArray: Array<keyof FormInterface>, errorKey = 'oneOf'): TypedValidatorFn<FormInterface> {
```

Comment:
Oh nvm I just saw why:

`NgxSubFormValidators.oneOf(['allianceRebel', 'allianceImperial'], 'oneOfAllianceType')`

Few things here, the key should probably be at the beginning (just a preference) but something that might lead to weird results:

When using 2 `oneOf` (or more), as the key is optional it would override the previous result if someone forgets to pass it.

I think it'd probably be better to have the definition of `oneOf` to be :

```
interface ControlInterface {
  randomKey: any;
  polyTypeA: string;
  polyTypeB: string;
}

interface FormInterface {
  subTypeA1: string;
  subTypeA2: string;
  subTypeB1: string;
  subTypeB2: string;
}

export function oneOf<ControlInterfaceType, FormInterfaceType>(
  keys: Partial<Record<keyof ControlInterface, FormInterface>>
) {
  // ...
}

oneOf<ControlInterface, FormInterface>({
  polyTypeA: ['subTypeA1', 'subTypeA2'],
  polyTypeB: ['subTypeB1', 'subTypeB2'],
});
```

This way, we only need to use `oneOf` once, it'll loop over the properties of the form once too (not X times where X is the number of times you call `oneOf`)

Also, in the above we'd be using as key what's defined in the original object and mapping that to keys of the sub form 

The above doesn't fully work though...

I get `Type 'string[]' is not assignable to type 'FormInterface'` but that should be doable :)

Reply:
The intent here, and what I’m optimising for is the common case. It strikes me as incredibly rare that you’d want to have multiple oneOfs in a single form, so I think we should not be making the api awkward for that one case.

I was also struggling with the problem of the error clobbering if you forgot to set the key name for differentiation, but I think you’ve got a great idea with having the argument be an object. However, I think we should support both options simultaneously, again to avoid an awkward interface for the extremely rare case. This could be handled by just checking if the first arg is array or object

Reply:
Another possibly simpler option is to have NgxSubFormValidators.oneOf and NgxSubFormValidators.oneOfMultiple